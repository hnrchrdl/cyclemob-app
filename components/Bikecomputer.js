import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import PropTypes from 'prop-types';

Bikecomputer.propTypes = {
  speed: PropTypes.number, // m/s
  altitude: PropTypes.number // m
};

function Bikecomputer({ speed, altitude }) {
  return (
    <View style={styles.container}>
      <View style={styles.displayItem}>
        <Text style={styles.displayItemDescription}>Speed</Text>
        <Text style={styles.displayItemValue}>
          {speed ? `${(((speed || 0) * 60 * 60) / 1000).toFixed(1)} km/h` : ''}
        </Text>
      </View>

      <View style={styles.displayItem}>
        <Text style={styles.displayItemDescription}>Altitude</Text>
        <Text style={styles.displayItemValue}>
          {altitude ? `${altitude.toFixed(1)} m` : '-'}
        </Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 0,
    paddingBottom: 0,
    flexDirection: 'row',
    opacity: 0.9,
    backgroundColor: '#fff'
  },
  displayItem: {
    padding: 10,
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  displayItemDescription: {
    fontSize: 10
  },
  displayItemValue: {
    fontSize: 14
  }
});

export default Bikecomputer;
