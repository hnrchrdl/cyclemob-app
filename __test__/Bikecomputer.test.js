import React from 'react';
import Bikecomputer from '../components/Bikecomputer';

import { shallow } from 'enzyme';

let props = {};

it('renders without crashing', () => {
  const tree = shallow(<Bikecomputer {...props} />);
  expect(tree).toBeTruthy();
});

it('matches snapshots', () => {
  const tree = shallow(<Bikecomputer {...props} />);
  expect(tree).toMatchSnapshot();
});

props = {
  speed: 10,
  altitude: 10
};

it('matches snapshot with new props', () => {
  const tree = shallow(<Bikecomputer {...props} />);
  expect(tree).toMatchSnapshot();
});
