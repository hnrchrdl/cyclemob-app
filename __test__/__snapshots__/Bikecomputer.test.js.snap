// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Bikecomputer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "padding": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 10,
              }
            }
          >
            Speed
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 14,
              }
            }
          >
            
          </Text>
        </Component>,
        <Component
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "padding": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 10,
              }
            }
          >
            Altitude
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 14,
              }
            }
          >
            -
          </Text>
        </Component>,
      ],
      "style": Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "flexDirection": "row",
        "opacity": 0.9,
        "paddingBottom": 2,
        "paddingTop": 2,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 10,
                }
              }
            >
              Speed
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 14,
                }
              }
            >
              
            </Text>,
          ],
          "style": Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "padding": 10,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Speed",
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 10,
              },
            },
            "ref": null,
            "rendered": "Speed",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "",
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 14,
              },
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 10,
                }
              }
            >
              Altitude
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 14,
                }
              }
            >
              -
            </Text>,
          ],
          "style": Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "padding": 10,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Altitude",
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 10,
              },
            },
            "ref": null,
            "rendered": "Altitude",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "-",
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 14,
              },
            },
            "ref": null,
            "rendered": "-",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "padding": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 10,
                }
              }
            >
              Speed
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 14,
                }
              }
            >
              
            </Text>
          </Component>,
          <Component
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "padding": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 10,
                }
              }
            >
              Altitude
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 14,
                }
              }
            >
              -
            </Text>
          </Component>,
        ],
        "style": Object {
          "backgroundColor": "#fff",
          "flex": 1,
          "flexDirection": "row",
          "opacity": 0.9,
          "paddingBottom": 2,
          "paddingTop": 2,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 10,
                  }
                }
              >
                Speed
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 14,
                  }
                }
              >
                
              </Text>,
            ],
            "style": Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "padding": 10,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Speed",
                "ellipsizeMode": "tail",
                "style": Object {
                  "fontSize": 10,
                },
              },
              "ref": null,
              "rendered": "Speed",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "",
                "ellipsizeMode": "tail",
                "style": Object {
                  "fontSize": 14,
                },
              },
              "ref": null,
              "rendered": "",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 10,
                  }
                }
              >
                Altitude
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 14,
                  }
                }
              >
                -
              </Text>,
            ],
            "style": Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "padding": 10,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Altitude",
                "ellipsizeMode": "tail",
                "style": Object {
                  "fontSize": 10,
                },
              },
              "ref": null,
              "rendered": "Altitude",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "-",
                "ellipsizeMode": "tail",
                "style": Object {
                  "fontSize": 14,
                },
              },
              "ref": null,
              "rendered": "-",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
